lib/PreviewProvider.js

```diff
@@ -0,0 +1,137 @@
+/* globals Task */
+"use strict";
+
+const {Cu} = require("chrome");
+const ss = require("sdk/simple-storage");
+const {setTimeout} = require("sdk/timers");
+const {siteFilter} = require("content-src/lib/filters");
+
+const EMBEDLY_PROXY_MAX_LINKS = 25; // number of links embedly proxy accepts per request
+const EMBEDLY_ENDPOINT = "https://embedly-proxy.dev.mozaws.net/v2/extract";
+const CLEAR_CACHE_TIMEOUT = 86400000; // 24 hours to clear cache
+
+Cu.importGlobalProperties(["fetch"]);
+Cu.import("resource://gre/modules/Task.jsm");
+
+function PreviewProvider() {
+  this.init();
+}
+
+PreviewProvider.prototype = {
+  // TODO: santize, filter, and dedupe urls
+  /**
+    * Clean up cache
+    */
+  cleanUpCache: Task.async(function*() {
```

Comment:
cleanUpCache will run 24 hours in the future after this function is called.
You actually intend to call this right away, and have a timeout function seperately, correct?